internal/translator/data/rules.go

```diff
@@ -34,6 +36,106 @@ type Rules struct {
 // RulesHelper has useful functions to work with the collection
 type RulesHelper interface {
 	FindByPattern(ctx context.Context, source, target, pattern string) ([]Rule, error)
+	NewRule(ctx context.Context, ruleStruct Rule) (*primitive.ObjectID, error)
+	FindRuleByPattern(ctx context.Context, sourceLanguage, targetLanguage, pattern string) ([]Rule, error)
+	UpdateOne(ctx context.Context, filter Rule, updateValue Rule) error
+	DeleteOne(ctx context.Context, rule Rule) error
+	DeleteMany(ctx context.Context, sourceLanguage, targetLanguage, pattern string) error
+}
+
+// NewRule creates a rule in database
+func (r Rules) NewRule(ctx context.Context, ruleStruct Rule) (*primitive.ObjectID, error) {
+	result, error := r.Collection.InsertOne(ctx, ruleStruct)
+
+	if error != nil {
+		return nil, error
+	}
+
+	newObjectID := result.InsertedID.(primitive.ObjectID)
+
+	return &newObjectID, nil
+}
+
+// FindRuleByPattern return a list of rules that match with the given pattern
+func (r Rules) FindRuleByPattern(ctx context.Context, sourceLanguage, targetLanguage, pattern string) ([]Rule, error) {
```

Comment:
`FindRule is redundant` Only `Find` is needed. Because you will do `Rule.FindRuleByPatter` do you see that you don't need the second `Rule` word? :+1:

Reply:
👍